File: Programming space/Pyhton Programs/BalancedBrackets.py
def balanced_bracket(brackets):
    stack = []
    bracket_map = {')':'(','}':'{',']':'['}
    for bracket in brackets:
        if bracket in bracket_map.values():
            stack.append(bracket)
        elif bracket in bracket_map.keys():
            if stack and stack[-1] == bracket_map[bracket]:
                stack.pop()
            else:
                return "Unbalanced Brackets"
    return "Balanced Brackets" if not stack else "Unbalanced Brackets"

if __name__ == "__main__":
    input_string = "{([])}"
    result  = balanced_bracket(input_string)
    print(result)

File: Programming space/Java Programs/BalancedBrackets.java
import java.util.*;
public class BalancedBrackets{

    public String brackerBalanceCheck(String iputString){
        Stack<Character> bracketStack = new Stack<>();
        Map<Character,Character> bracketMap = Map.of(')','(',']','[','}','{');
        for(char bracket: bracketStack){
            if(bracketMap.containsValue(bracket)){
                bracketStack.add(bracket);
            }
            else if(bracketMap.containsKey(bracket)){
                if(!bracketStack.isEmpty() && bracketStack.peek().equals(bracketMap.get(bracket)))
                    bracketStack.pop();
                else
                    return "Unbalanced Brackets";
            }
        }
        if(bracketStack.isEmpty())
            return "Balanced Brackets";
        else
            return "Unbalanced Brackets";
    }

  public static void main(String[] args) {
    BalancedBrackets checker = new BalancedBrackets();
    String brackets = "{[()]}";
    String result = checker.brackerBalanceCheck(brackets);
    System.out.println(result);
  }
}

  
